### DISCUSSION TITLE
--forceupdate doesnt work multithreaded.

### DISCUSSION THREAD

**Discussion by kgncengiz** [PaperMC/Paper#10602] [Ideas]
### Is your feature request related to a problem?

--forceupdate doesn't work multithreaded and it's too slow for updating big worlds. updateing 15k*15k world took 30 minutes.

### Describe the solution you'd like.

make the world updater update chunks parallel not sequentially.

### Describe alternatives you've considered.

there is no alternatives.

### Other

have a good day.

**Comment by electronicboy** [comment]
It already is threaded, but the math for determining cores could maybe be less restrictive
`int threads = Runtime.getRuntime().availableProcessors() * 3 / 8;`

**Comment by Spottedleaf** [comment]
It may be worthwhile to add a startup flag to control the number of threads.

**Comment by kgncengiz** [comment]
> It already is threaded, but the math for determining cores could maybe be less restrictive `int threads = Runtime.getRuntime().availableProcessors() * 3 / 8;` 



I thing math needs to be improved becouse i give 8 cores to the folia for test and it peaked at 400%(1 core=100%) its half the max speed of my hardware. Btw when i give 16 cores it peaked at 1000%(1 core=100%).
